torch/csrc/jit/passes/onnx/fixup_onnx_controlflow.cpp

```diff
@@ -258,25 +259,41 @@ bool IsUninitializedNode(Node* n) {
 
 // Infer shape and type of the uninitialized_output from the corresponding
 // output of the other subblock. prim::Uninitialized node is proven to be
-// unused. So replace this node with a constant of the inferred shape and type.
+// unused. So replace this node with a constant(TensorType) or
+// Sequence(ListType) of the inferred shape and type.
 void InferShapeTypeForUninitializedOutput(
     Graph* graph,
     Block* block,
     Value* uninitialized_output,
     Value* other_output) {
-  auto output_type = other_output->type()->expect<TensorType>();
-  auto elem_type = at::initialTensorOptions().dtype(output_type->scalarType());
-  Node* const_node = graph->create(::c10::onnx::Constant, 1);
-
-  if (output_type->sizes().concrete_sizes().has_value()) {
-    auto size = output_type->sizes().concrete_sizes().value();
-    const_node->t_(attr::value, at::zeros(size, elem_type));
-    const_node->output()->setType(other_output->type());
-    const_node->output()->copyMetadata(other_output);
-  } else {
-    const_node->t_(attr::value, at::zeros({}, elem_type));
-    const_node->output()->setType(
-        TensorType::create(*(output_type->scalarType()), at::kCPU, {}, {}));
+  Node* const_node = nullptr;
+  if (auto output_type = other_output->type()->cast<TensorType>()) {
+    auto elem_type =
+        at::initialTensorOptions().dtype(output_type->scalarType());
+    const_node = graph->create(::c10::onnx::Constant, 1);
+
+    if (output_type->sizes().concrete_sizes().has_value()) {
+      auto size = output_type->sizes().concrete_sizes().value();
+      const_node->t_(attr::value, at::zeros(size, elem_type));
+      const_node->output()->setType(other_output->type());
+    } else {
+      const_node->t_(attr::value, at::zeros({}, elem_type));
+      const_node->output()->setType(
+          TensorType::create(*(output_type->scalarType()), at::kCPU, {}, {}));
+    }
+  } else if (auto output_type = other_output->type()->cast<ListType>()) {
+    TypePtr elem = output_type->getElementType();
+    const_node = graph->create(::c10::onnx::SequenceEmpty, 1);
```

Comment:
Where is the code you add Identity node? Once you insert a new node to graph, you may need add shape inference for that node. See https://github.com/pytorch/pytorch/blob/dec5aa2260cef540b622bd9a9504b6f11cb1f607/torch/csrc/jit/passes/onnx.cpp#L286

Reply:
thanks David！
I'm not sure if SequenceEmpty needs an certain shape, will check in ONNX spec and try to use this function.

Reply:
btw, add Identity node in thins link [fixup_onnx_controlflow.cpp#L190]. We only deal with the TensorList case when Infer Shape and Type For UninitializedOutput.(https://github.com/pytorch/pytorch/blob/e523674c51072036a69125f098eb16c93f32b59c/torch/csrc/jit/passes/onnx/fixup_onnx_controlflow.cpp#L190)